include/wex/presets/Component.hpp

```diff
@@ -0,0 +1,30 @@
+#include "../Common.hpp"
+#include <SFML/System/Vector2.hpp>
+
+namespace wex {
+
+/// \brief A component represents a bag of data.
+/// A component by itself has no meaning, and is useful to describe properties or attributes
+/// of a `wex::GameObject`.
+/// Extend this class to use custom components.
+struct Component {
```

Comment:
Convention is to have such things be `class`es, `struct`s don't usually have virtual semantics.

Reply:
My reasoning was this is more of a C-style "bag of data", rather than mutable state + mutating functions. So I went with a struct here. is that okay?

Reply:
Sure, just letting you know what to expect elsewhere